src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -118,7 +118,7 @@ public function validate($value, Constraint $constraint)
      */
     private function checkMX($host)
     {
-        return checkdnsrr($host, 'MX');
+        return ('' !== $host) && checkdnsrr($host, 'MX');
```

Comment:
Well there is a similar check line 45 `if (null === $value || '' === $value)` that does not use the native empty function so I assumed I should do the same but maybe I was wrong ?